### PR TITLE
[#4910] Add examples for `Style/ClassVars` cop

### PR DESCRIPTION
**Pull request by walf443**
related #4910 

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `rake default` or `rake parallel`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by walf443** [comment]
It maybe better to write more doc about why using class variable is not good.

**Comment by bbatsov** [comment]
> It maybe better to write more doc about why using class variable is not good.

Certainly. It might be nice to also show can people are supposed to access the class instance variables, as this might not be obvious to everyone.

**Comment by walf443** [comment]
@bbatsov I updated document. how about it?

File: lib/rubocop/cop/style/class_vars.rb [hunk 1]
```diff
@@ -6,6 +6,29 @@ module Style
...
+      #   end
+      #
+      #   # good
+      #   class A
+      #     @test = 10
+      #   end
+      #
+      #   class A
+      #     def test
+      #       @@test # you can access class variable without offence
```
Comment (vtpros) [inline lib/rubocop/cop/style/class_vars.rb:28]: This example gives an impression that by using @@test here you can access class-level instance vabiable set on line 23, which is not the case. Also, to access class variable without offence here you need to set it first, no? And that would be an offence to rubycop.

**Pull request merged**